crates/precompile/src/secp256k1.rs

```diff
@@ -57,6 +61,16 @@ mod secp256k1 {
     }
 }
 
+#[cfg(not(any(feature = "k256", feature = "secp256k1")))]
+mod secp256k1 {
+    use crate::B256;
+
+    #[deprecated = "warning: no `ecRecover` backend is enabled"]
+    pub fn ecrecover(_sig: &[u8; 65], _msg: &B256) -> Result<B256, ()> {
+        unimplemented!("no `ecRecover` backend is enabled")
+    }
+}
```

Comment:
LMK if you want to make this a hard compile error. I made both ecrecover backends optional @rakita